### PR TITLE
Initial commit of booleanPointOnLine

### PR DESCRIPTION
**Pull request by rowanwins**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Have read [How To Contribute](https://github.com/Turfjs/turf/blob/master/CONTRIBUTING.md#how-to-contribute).
- [x] Run `npm test` at the sub modules where changes have occurred.
- [x] Run `npm run lint` to ensure code style at the turf module level.

- [x] Overview description of proposed module.
- [x] Include JSDocs with a basic example.
- [x] Execute `./scripts/generate-readmes` to create `README.md`.
- [x] Add yourself to **contributors** in `package.json` using "Full Name <@GitHub Username>".

## booleanPointOnLine
Returns true if a point is on a line. Accepts a optional parameter to ignore the start and end vertices of the linestring.

This is a snippet that we've used across a number of the boolean modules now so figured it was worth separating out.

### CODE REVIEW

**Comment by stebogit** [comment]
@rowanwins @DenisCarriere 
looking at this new module (:+1: btw), may I suggest (for the following major release) to rename `@turf/point-on-line` to something like `@turf/nearest-point-on-line`?
Otherwise it's a little confusing and the user _has_ to go to the docs or the code to understand the difference.
We might even "group" the `nearest` modules, allowing an easy addition of other modules alike, renaming also `@turf/nearest` to `@turf/nearest-point`

**Comment by DenisCarriere** [comment]
> @turf/point-on-line to something like @turf/nearest-point-on-line?
Otherwise it's a little confusing and the user has to go to the docs or the code to understand the difference.
We might even "group" the nearest modules, allowing an easy addition of other modules alike, renaming also @turf/nearest to @turf/nearest-point

👍 I like this change for a next major release, it is a big confusing when they aren't grouped and the name is too generalized.

File: packages/turf-boolean-point-on-line/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * @example
+ * var pt = turf.point([0, 0]);
+ * var line = turf.lineString([[-1, -1],[1, 1],[1.5, 2.2]]);
+ * var isPointOnLine = turf.booleanPointOnLine(pt, line);
+ * //=true
+ */
+module.exports = function (point, linestring, ignoreEndVertices) {
+    var pointCoords = getCoords(point);
+    var lineCoords = getCoords(linestring);
+    for (var i = 0; i < lineCoords.length - 1; i++) {
```
Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/index.js]: 👍 Code looks good.

We really need a `segmentEach` module (also mentioned here https://github.com/Turfjs/turf/pull/856#discussion_r128902548)

File: packages/turf-boolean-point-on-line/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+var getCoords = require('@turf/invariant').getCoords;
+
+/**
+ * Returns true if a point is on a line. Accepts a optional parameter to ignore the start and end vertices of the linestring.
+ *
+ * @name booleanPointOnLine
+ * @param {Geometry|Feature<Point>} point GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<LineString>} linestring GeoJSON Feature or Geometry
+ * @param {Boolean} [ignoreEndVertices=false] whether to ignore the start and end vertices.
+ * @returns {Boolean} true/false
```
Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/index.js]: I remember having this conversation with @tmcw about `Boolean` & `boolean`.

From what I can read on JSDocs, it's saying to use `boolean`
http://usejsdoc.org/tags-type.html

Also DocumentationJS linting flags `Boolean` as a warning (which we will be adding soon)

This change applies to many modules, so don't need to fix this right now.

File: packages/turf-boolean-point-on-line/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/// <reference types="geojson" />
+
+type Feature = GeoJSON.Feature<any> | GeoJSON.GeometryObject;
+
+/**
+ * http://turfjs.org/docs/#booleanPointOnLine
```
Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/index.d.ts]: All lowercase for the URL.

Example http://turfjs.org/docs/#centerOfMass (doesn't work)

File: packages/turf-boolean-point-on-line/bench.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Benchmark Results
+ *
+ * PointOnLineEnd: 0.355ms
+ * PointOnLineMidpoint: 0.019ms
+ * PointOnLineStart: 0.021ms
+ * PointOnLineEnd x 13,957,263 ops/sec ±0.53% (91 runs sampled)
+ * PointOnLineMidpoint x 17,388,052 ops/sec ±0.46% (94 runs sampled)
+ * PointOnLineStart x 17,036,405 ops/sec ±1.34% (91 runs sampled)
```
Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/bench.js]: 🏎  💨 Zoom Zoom!

File: packages/turf-boolean-point-on-line/index.js (hunk 1 above)
Comment (rowanwins) [inline packages/turf-boolean-point-on-line/index.js]: The only thing I find tricky about the invariant type modules is a lack of flow control, but in a lot of cases they are super-handy!

Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/index.js]: What do you mean by lack of flow control?

File: packages/turf-boolean-point-on-line/index.d.ts [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
 type Point = GeoJSON.Feature<GeoJSON.Point> | GeoJSON.Point;
-type LineString = GeoJSON.LineString;
-type Line = GeoJSON.Feature<LineString> | LineString;
+type Line = GeoJSON.Feature<GeoJSON.LineString> | GeoJSON.LineString;
 
 /**
  * http://turfjs.org/docs/#booleanpointonline
  */
-declare function booleanPointOnLine(feature1: Point, feature2: Line, ignoreEndVertices: boolean): boolean;
+declare function booleanPointOnLine(point: Point, linestring: Line, ignoreEndVertices?: boolean): boolean;
```
Comment (DenisCarriere) [inline packages/turf-boolean-point-on-line/index.d.ts]: @rowanwins Another thing with these definitions, I try to use the same parameter names as the `index.js`

And `ignoreEndVertices` is optional, you can add `?` to define optional params.

**Pull request merged**